add Vehicle Namespace

tylerreichle opened the pull request:
# Vehicle Generator

- vehicle: returns random make/model combination
- manufacturer: random manufacturer
- model: random vehicle model
- type: random vehicle type
- fuel: random fuel type
- vin: random valid vin number
- color: random color

# random.alpha
 - generate random alpha characters.
 - `alpha()` returns single lowercase letter
 - `alpha({upcase: true})` returns single uppercase letter
 - `alpha({count: 5, upcase: true})` returns five uppercase letters

includes unit tests for all functions

tylerreichle commented:
#506

Marak commented:
@tylerreichle -

This is starting to look good. I notice at least one small code error, but overall I like.

Let me know when you are finished and I will review and merge.

Marak commented on lib/vehicle.js:
```diff
@@ -0,0 +1,109 @@
+  /**
+   * manufacturer
+   *
+   * @method faker.database.manufacturer
```
I think this may be a typo?

tylerreichle replied:
Wow we must have noticed the typo at the same time! I just corrected 😄

Marak commented on lib/locales/en/vehicle/vin.js:
```diff
@@ -0,0 +1,32 @@
+module["exports"] = [
+  "YV1MH682762184654",
```
Are these all legitimate VIN numbers?

I tried looking up a few and they appear to be returning information for real cars.

If so, that could cause issue to owners of vehicle somehow.

tylerreichle replied:
These came from an online vin generator that was using regex to return valid vins. I assumed users wanting to generate random vins would have a similar check to ensure they are valid.

Wonder how we can deal with generating valid vins without returning one in use?

tylerreichle replied:
I could update to have vins returned from the included helper regexpStyleStringParse. That way no real vins will be hardcoded

Marak replied:
It would be better I think if we could port the regex / code for generating VINs and place that in the API method.

I'm OK with generating valid VINs from algorithm. A pre-defined list seems bad because we would be singling out specific vehicles ( which could end up bad ).

Marak replied:
That sounds good.

Let's see if we can remove the list of hard-coded VINs in favor of algo.

Would be nice if we could also have simple test, perhaps cross-reference documentation in comments for format of VIN specification.

Thanks!

tylerreichle replied:
Yeah, I definitely see the problem now that you point it out! I will make those changes and let you know when ready for review

tylerreichle commented:
@Marak I updated vin to return using algorithm instead of hardcoded options. I also added random.alpha function for character generation in vin.

I also ended up using random.alpha in new airport namespace PR I just raised 👍

tilomitra commented:
Can we go ahead and merge this? It's just sitting here for 9 months.

Marak commented:
Marked for inclusion in next release `v5`

Marak commented:
Merged.

Thank you for your contributions @tylerreichle 

Closing.

pkuczynski commented:
Still not published? See #760
